utilities/ride-model-util/src/main/java/com/adobe/ride/utilities/model/ModelObject.java

```diff
@@ -1457,53 +1624,126 @@ protected String getRelativeResourceLocation(String controlPath, String relative
+    Object exisitingValue = null;
+    if (path == null || path == null + "/" + null) {
+      exisitingValue = null;
+    } else {
+      try {
+        String jsonstring = "";
+
+        if (getModelType() == ModelPropertyType.OBJECT) {
+          jsonstring = objectMetadata.toJSONString();
+        } else if (getModelType() == ModelPropertyType.ARRAY) {
+          jsonstring = objectItems.toJSONString();
+        }
+
+        // look up value (strip trailing slash from parent path)
+        Object value = mapper.readTree(jsonstring).at(path.replaceAll("/$", "")).get(key);
+
+        if (value != null) {
+          exisitingValue = (Object) value;
+        } else {
+          exisitingValue = null;
+        }
+      } catch (Exception e) {
+        logger.log(Level.SEVERE, e.getMessage());
+      }
+    }
+    return exisitingValue;
+  }
+
   /**
    * Method to cast and post value to json node path.
    * 
-   * @param type ModelPropertyType
-   * @param pathToParent String Fully qualified path to parent node of the target
-   * @param key String target node
+   * @param metadataPath String Fully qualified path to parent node of the target
+   * @param key String target node key
    * @param value Object value to be assigned
    */
-  protected void setMetadataValue(ModelPropertyType type, String pathToParent, String key,
-      Object value) {
+  public void setMetadataValue(String metadataPath, String key, Object value) {
     JsonNode tree = null;
-    pathToParent = (pathToParent == "/") ? "" : pathToParent;
+
     try {
 
-      tree = mapper.readTree(objectMetadata.toJSONString());
+      tree = mapper.readTree(getMetadataString());
     } catch (IOException e) {
       logger.log(Level.SEVERE, e.getMessage());
     }
-    if (type == ModelPropertyType.BOOLEAN) {
-      ((ObjectNode) tree.at(pathToParent)).put(key, Boolean.parseBoolean(value.toString()));
-    } else if (type == ModelPropertyType.INTEGER) {
-      ((ObjectNode) tree.at(pathToParent)).put(key, Long.parseLong(value.toString()));
-    } else if (type == ModelPropertyType.NUMBER) {
-      ((ObjectNode) tree.at(pathToParent)).put(key, Double.parseDouble(value.toString()));
-    } else if (type == ModelPropertyType.OBJECT || type == ModelPropertyType.REF_DEFINITION
-        || type == ModelPropertyType.REF_SCHEMA || type == ModelPropertyType.ARRAY) {
+
+    String path = metadataPath.replaceAll("/$", "");
+
+    String stringRep = castAsString(value);
+    if (isBoolean(value)) {
+      ((ObjectNode) tree.at(path)).put(key, Boolean.parseBoolean(stringRep));
+    } else if (isInt(value)) {
+      ((ObjectNode) tree.at(path)).put(key, Long.parseLong(stringRep));
+    } else if (isNumber(value)) {
+      ((ObjectNode) tree.at(path)).put(key, Double.parseDouble(stringRep));
+    } else if (isArrayNode(value)) {
+      ((ObjectNode) tree.at(path)).putArray(key).addAll((ArrayNode) value);
+    } else if (isObject(value)) {
```

Comment:
Updated these conditionals to set a property value based on the object that is being passed in, not on the definition of the property.